libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py

```diff
@@ -637,7 +652,7 @@ def get_available_models(
         # nim model listing does not provide the type and we cannot know
         # the model name ahead of time to guess the type.
         # so we need to list all models.
-        if mode == "nim":
+        if mode == "nim" or not self.is_hosted:
```

Comment:
Can't you use the /v1/models API here? 
https://developer.nvidia.com/docs/nemo-microservices/inference/reference/openai-api.html

Reply:
yes. there's another wip change that migrates off nvcf and uses integate.api.nvidia.com/v1/models.